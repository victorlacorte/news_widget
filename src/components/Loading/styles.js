import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const Ring = styled.div`
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
`;

const RingChild = styled.div`
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 51px;
  height: 51px;
  margin: 6px;
  border-width: 6px;
  border-style: solid;
  border-radius: 50%;
  border-color: ${({ theme }) => theme.palette.blue.main} transparent transparent transparent;
  animation: ${rotate} 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;

  &:nth-child(1) {
    animation-delay: -0.45s;
  }

  &:nth-child(2) {
    animation-delay: -0.3s;
  }

  &:nth-child(3) {
    animation-delay: -0.15s;
  }
`;

export {
  Ring,
  RingChild,
};
