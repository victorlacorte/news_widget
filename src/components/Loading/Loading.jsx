import React from 'react';

import { Ring, RingChild } from './styles';

function Loading() {
  return (
    <Ring>
      <RingChild />
      <RingChild />
      <RingChild />
      <RingChild />
    </Ring>
  );
}

export default Loading;
