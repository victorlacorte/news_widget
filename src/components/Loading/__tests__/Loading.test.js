import React from 'react';
import { shallow } from 'enzyme';

import Loading from '../Loading';

describe('<Loading />', () => {
  it('creates a Loading component', () => {
    const wrapper = shallow(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
