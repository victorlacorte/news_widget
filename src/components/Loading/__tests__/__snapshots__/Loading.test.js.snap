// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> creates a Loading component 1`] = `
<styled.div>
  <styled.div />
  <styled.div />
  <styled.div />
  <styled.div />
</styled.div>
`;
